packaging/os/yum.py

```diff
@@ -603,6 +603,33 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
         cmd = yum_basecmd + ['install'] + pkgs
 
         if module.check_mode:
+            ts = yum.rpm.TransactionSet()
+            ts.setVSFlags(yum.rpm._RPMVSF_NOSIGNATURES)
+            yb = yum_base()
+
+            for package in pkgs:
+                if package.startswith('http'):
+                    tmp_file = tempfile.TemporaryFile()
+
+                    try:
+                        tmp_file, info = fetch_url(module, pkg)
+                        tmp_file.write(rsp.read())
+                    except Exception, e:
+                        module.fail_json(msg="Error downloading %s: %s" % (package, e))
+
+                    tmp_file.seek(0)
+
+                elif os.path.exists(package):
+                    tmp_file = os.open(package, os.O_RDONLY)
+
```

Comment:
I think we need one more else: here to handle the case where package is a package name rather than a file or url.